cpp/server/ct-server.cc

```diff
@@ -202,11 +202,15 @@ void SequenceEntries(TreeSigner<LoggedCertificate>* tree_signer,
 
   while (true) {
     if (election->IsMaster()) {
-      store->CleanupOldEntries();
-
-      // TODO(alcutter): Probably don't need to blow up here
-      CHECK_EQ(util::Status::OK, tree_signer->SequenceNewEntries())
-          << "Problem sequencing new entries";
+      util::Status status(store->CleanupOldEntries());
+      if (!status.ok()) {
+        LOG(WARNING) << "Problem cleaning up old entries: " << status;
+      } else {
+        status = tree_signer->SequenceNewEntries();
```

Comment:
I figured that if the clean-up failed it's likely because of some transient problem (or at least, that's the best possible explanation!), so going back and hitting etcd again straight away was likely to also fail and so best to skip the rest of the run entirely.

I don't feel strongly about it though, so feel free to convince me...